### PR TITLE
style: Use f-strings wherever possible

### PR DESCRIPTION
**Pull request by DimitriPapadopoulos**
They're faster.

### CODE REVIEW

File: tests/test_spec_examples.py [hunk 1]
```diff
@@ -39,8 +39,7 @@
 #             text = text.replace('\u21d4', '')    # byte order mark
 #             text = text.replace('\u2192', '\t')  # right arrow
 #             text = text.replace('\u00b0', '')    # empty scalar
-#             with open('tests/yaml-1.2-spec-examples/%s' % id, 'w',
-#                       encoding='utf-8') as g:
```
Comment (DimitriPapadopoulos) [inline tests/test_spec_examples.py:43]: Note that `encoding='utf-8'` is the default.

File: yamllint/config.py [hunk 2]
```diff
@@ -179,16 +179,16 @@ def validate_rule_conf(rule, conf):
...
             # Example: CONF = {option: (bool, 'mixed')}
             #          → {option: true}         → {option: mixed}
             if isinstance(options[optkey], tuple):
                 if (conf[optkey] not in options[optkey] and
                         type(conf[optkey]) not in options[optkey]):
                     raise YamlLintConfigError(
-                        'invalid config: option "%s" of "%s" should be in %s'
-                        % (optkey, rule.ID, options[optkey]))
+                        f'invalid config: option "{optkey}" of "{rule.ID} '
+                        f'should be in {options[optkey]}')
```
Comment (adrienverge) [inline yamllint/config.py:191]: There's a missing `"` after `{rule.ID}`, making the tests fail:

```
FAIL: test_yes_no_for_booleans (tests.test_config.SimpleConfigTestCase)
----------------------------------------------------------------------
yamllint.config.YamlLintConfigError: invalid config: option "indent-sequences" of "indentation should be in (<class 'bool'>, 'whatever', 'consistent')

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/runner/work/yamllint/yamllint/tests/test_config.py", line 109, in test_yes_no_for_booleans
    c = config.YamlLintConfig('rules:\n'
AssertionError: "invalid config: option "indent-sequences" of "indentation" should be in " does not match "invalid config: option "indent-sequences" of "indentation should be in (<class 'bool'>, 'whatever', 'consistent')"
```

**Review by adrienverge — COMMENTED** [review]
Looks good, thanks for taking the time!

Have you changed them one by one and checked they all make sense? Or did you use a script to find and replace them? (in some rare cases like `%02d`, `%.2f` or `%-10s`, f-string cannot be used as is)

File: yamllint/config.py (hunk 2 above)
Comment (DimitriPapadopoulos) [reply yamllint/config.py:191]: You're right about UTF-8, I will revert that:
https://docs.python.org/3/library/functions.html#open

Comment (DimitriPapadopoulos) [reply yamllint/config.py:191]: Let me have a look at the rest of the formatting.

File: tests/test_spec_examples.py (hunk 1 above)
Comment (DimitriPapadopoulos) [reply tests/test_spec_examples.py:43]: You're right about UTF-8, I will revert that:
https://docs.python.org/3/library/functions.html#open

**Comment by DimitriPapadopoulos** [comment]
> Have you changed them one by one and checked they all make sense? Or did you use a script to find and replace them? (in some rare cases like `%02d`, `%.2f` or `%-10s`, f-string cannot be used as is)

There are not such cases as far as I can see. Yet, it possible to write `f{variable!02d}` or `f{variable!.2f}`, isn't it?

**Comment by adrienverge** [comment]
> There are not such cases as far as I can see. Yet, it possible to write `f{variable!02d}` or `f{variable!.2f}`, isn't it?

Sure it is! I just wanted to make sure you (or your custom script?) looked for them, if any :)

I still see a few `%s` and `%d` in some files (namely `yamllint/linter.py`, `yamllint/rules/line_length.py`, `yamllint/rules/indentation.py` and `yamllint/rules/octal_values.py`). Could you check them?

**Comment by DimitriPapadopoulos** [comment]
As for the tool I use, it's [pyupgrade](https://github.com/asottile/pyupgrade). But I often end up applying changes manually, because:
1. You cannot limit `pyupgrade` to a single class of changes. I prefer an individual commit per class of changes.
2. It's not perfect: it often leaves behind some cases of `%` string interpolation.

**Comment by DimitriPapadopoulos** [comment]
I have left an instance of `%` string interpolation in `yamllint/rules/line_length.py` because I find the f-string interpolation is less readable – unless of course I use additional variables to store the result of complex expressions.

**Review by adrienverge — APPROVED** [review]
> I have left an instance of `%` string interpolation in `yamllint/rules/line_length.py`

I had spotted this one too and I agree it's better readable like this :+1: 

I allowed myself to force-push to your branch with a slight rework of modified strings:
```diff
--- a/yamllint/config.py
+++ b/yamllint/config.py
@@ -179,8 +179,8 @@ def validate_rule_conf(rule, conf):
                 continue
             if optkey not in options:
                 raise YamlLintConfigError(
-                    f'invalid config: unknown option "{optkey}" '
-                    f'for rule "{rule.ID}"')
+                    f'invalid config: unknown option "{optkey}" for rule '
+                    f'"{rule.ID}"')
             # Example: CONF = {option: (bool, 'mixed')}
             #          → {option: true}         → {option: mixed}
             if isinstance(options[optkey], tuple):
@@ -216,8 +216,8 @@ def validate_rule_conf(rule, conf):
                 raise YamlLintConfigError(f'invalid config: {rule.ID}: {res}')
     else:
         raise YamlLintConfigError(
-            f'invalid config: rule "{rule.ID}": should be '
-            f'either "enable", "disable" or a dict')
+            f'invalid config: rule "{rule.ID}": should be either "enable", '
+            f'"disable" or a dict')
 
     return conf
 
--- a/yamllint/rules/indentation.py
+++ b/yamllint/rules/indentation.py
@@ -342,11 +342,11 @@ def _check(conf, token, prev, next, nextnext, context):
 
         if found_indentation != expected:
             if expected < 0:
-                message = f'wrong indentation: expected ' \
-                          f'at least {found_indentation + 1}'
+                message = f'wrong indentation: expected at least ' \
+                          f'{found_indentation + 1}'
             else:
-                message = f'wrong indentation: expected {expected} ' \
-                          f'but found {found_indentation}'
+                message = f'wrong indentation: expected {expected} but ' \
+                          f'found {found_indentation}'
             yield LintProblem(token.start_mark.line + 1,
                               found_indentation + 1, message)
 
```

Thank you Dimitri, let's merge!

**Pull request merged**